pkg/sriov/sriov.go

```diff
@@ -207,19 +220,112 @@ func (s *sriovManager) ReleaseVF(conf *types.NetConf, podifName, cid string, net
 		if err = s.nLink.LinkSetNsFd(linkObj, int(initns.Fd())); err != nil {
-			return fmt.Errorf("failed to move interface %s to init netns: %v", conf.HostIFNames, err)
+			return fmt.Errorf("failed to move interface %s to init netns: %v", podifName, err)
 		}
 
 		return nil
 	})
+
+	// If we skipped renaming due to missing host interface name, rebind the VF to get correct name
+	if err == nil && useRebindForNaming {
+		// Rebind the VF to ensure it gets the correct interface name in host namespace
+		_ = s.utils.RebindVf(conf.Master, conf.DeviceID)
+		// Don't fail the entire operation if rebind fails - the interface is already back in host namespace
+		// This is just for getting the correct name, which is not critical for deletion success
+	}
+
+	return err
+}
+
+// setVFLinkState sets the VF link state
+func (s *sriovManager) setVFLinkState(conf *types.NetConf, pfLink netlink.Link) error {
+	if conf.LinkState == "" {
+		return nil
+	}
+
+	var state uint32
+	switch conf.LinkState {
+	case "auto":
+		state = netlink.VF_LINK_STATE_AUTO
+	case "enable":
+		state = netlink.VF_LINK_STATE_ENABLE
+	case "disable":
+		state = netlink.VF_LINK_STATE_DISABLE
+	default:
+		// the value should have been validated earlier, return error if we somehow got here
+		return fmt.Errorf("unknown link state %s when setting it for vf %d", conf.LinkState, conf.VFID)
+	}
+
+	if err := s.nLink.LinkSetVfState(pfLink, conf.VFID, state); err != nil {
+		return fmt.Errorf("failed to set vf %d link state to %d: %v", conf.VFID, state, err)
+	}
+
+	return nil
+}
+
+// handleVFGuidConfiguration handles VF GUID setting and validation
+func (s *sriovManager) handleVFGuidConfiguration(conf *types.NetConf, pfLink netlink.Link) error {
+	if conf.GUID != "" {
+		return s.configureVFGuid(conf, pfLink)
+	}
+	return s.validateVFGuid(conf)
+}
+
+// configureVFGuid configures the VF GUID, saving the current GUID for later restoration
+func (s *sriovManager) configureVFGuid(conf *types.NetConf, pfLink netlink.Link) error {
+	if !utils.IsValidGUID(conf.GUID) {
+		return fmt.Errorf("invalid guid %s", conf.GUID)
+	}
+
+	// For VFIO VF devices, we don't have a network interface, so skip VF link operations
+	if conf.VfioPciMode && conf.HostIFNames == "" {
+		// VFIO VF: Just set the GUID directly via PF, no VF link to query
+		return s.setVfGUID(conf, pfLink, conf.GUID)
+	}
+
+	// Normal VF (not VFIO): save current link guid and set new one
+	vfLink, err := s.nLink.LinkByName(conf.HostIFNames)
+	if err != nil {
+		return fmt.Errorf("failed to lookup vf %q: %v", conf.HostIFNames, err)
+	}
+
+	conf.HostIFGUID = vfLink.Attrs().HardwareAddr.String()[36:]
+
+	// Set link guid
+	return s.setVfGUID(conf, pfLink, conf.GUID)
+}
+
+// validateVFGuid validates that the VF has a valid GUID
+func (s *sriovManager) validateVFGuid(conf *types.NetConf) error {
+	// For VFIO VF devices, skip GUID validation since we can't access the VF interface
+	if conf.VfioPciMode && conf.HostIFNames == "" {
```

Comment:
why do we need the second condition ? when will we have empty HostIFNames but VfioPciMode is false